src/adapters/axiosAdapter.ts

```diff
@@ -13,12 +13,14 @@ export class AxiosAdapter<TResource> implements Adapter<TResource> {
                 method: options.method as Method,
                 data: options.requestBody,
                 headers: {
-                    "X-Octopus-ApiKey": options.configuration.apiKey
+                    "X-Octopus-ApiKey": options.configuration.apiKey ?? ''
                 },
                 responseType: "json", 
             };
             if (typeof XMLHttpRequest === 'undefined') {
-                config.headers["User-Agent"] = "ts-octopusdeploy";
+                if (config.headers) {
+                    config.headers["User-Agent"] = "ts-octopusdeploy";
+                }
```

Comment:
We need to incorporate a version number into this string for telemetry. Ideally, the string would look like `<client>/<version> (os-version)`.

Reply:
Created an issue for it @jbristowe 
https://github.com/OctopusDeploy/api-client.ts/issues/26